code/modules/paperwork/fax.dm

```diff
@@ -0,0 +1,133 @@
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "fax"
+	density = TRUE
+	power_channel = AREA_USAGE_EQUIP
+	max_integrity = 300
+	pass_flags = PASSTABLE
+	/// The unique ID by which the fax will build a list of existing faxes.
+	var/fax_id
+	/// The name of the fax displayed in the list. Not necessarily unique to some EMAG jokes.
+	var/fax_name
+	/// A reference to an `/obj/item/paper` inside the copier, if one is inserted. Otherwise null.
+	var/obj/item/paper/paper_contain
+	/// Necessary to hide syndicate faxes from the general list
+	var/syndicate_network = FALSE
+
+/obj/machinery/fax/Initialize(mapload)
+	. = ..()
+	if (!fax_id)
+		fax_id = SSnetworks.assign_random_name()
+	if (!fax_name)
+		fax_name = "Unregistered fax " + fax_id
+
+/obj/machinery/fax/Destroy()
+	QDEL_NULL(paper_contain)
+	return ..()
+
+/obj/machinery/fax/update_icon_state()
+	if(paper_contain)
+		icon_state = "fax_contain"
+	else
+		icon_state = "fax"
+	return ..()
+
+/obj/machinery/fax/emag_act(mob/user)
+	if(!(obj_flags & EMAGGED))
+		obj_flags |= EMAGGED
+		playsound(src, 'sound/creatures/dog/growl2.ogg', 50, FALSE)
+		to_chat(user, span_warning("An image appears on [src] screen for a moment with Jan in the cap of a Syndicate officer."))
+
+/obj/machinery/fax/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+	default_unfasten_wrench(user, tool)
+	return TOOL_ACT_TOOLTYPE_SUCCESS
+
+/obj/machinery/fax/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/paper))
+		if(!paper_contain)
+			paper_contain = I
+			I.forceMove(src)
+			update_appearance()
+		return
+	if(I.tool_behaviour == TOOL_MULTITOOL)
+		var/new_fax_name = tgui_input_text(user, "Enter a new name for the fax machine.", "New Fax Name", , 128)
+		if(!new_fax_name)
+			to_chat(user, span_warning("Invalid text!"))
+			return
+		if (new_fax_name != fax_name)
+			if (fax_name_exist(new_fax_name))
+				// Being able to set the same name as another fax machine will give a lot of gimmicks for the traitor.
+				if (syndicate_network != TRUE && obj_flags != EMAGGED)
+					to_chat(user, span_warning("There is already a fax machine with this name on the network."))
+					return
+			fax_name = new_fax_name
+		return
+	..()
+
+/obj/machinery/fax/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Fax")
+		ui.open()
+
+/obj/machinery/fax/ui_data(mob/user)
+	var/list/data = list()
+	var/faxes[0]
+	for(var/obj/machinery/fax/FAX in GLOB.machines)
+		// To avoid putting yourself on the list
+		if(FAX.fax_id != fax_id)
+			if(FAX.syndicate_network != TRUE)
+				faxes.Add(list(list("id" = FAX.fax_id, "name" = FAX.fax_name)))
+			else if (syndicate_network == TRUE || obj_flags & EMAGGED)
+				faxes.Add(list(list("id" = FAX.fax_id, "name" = FAX.fax_name)))
+
+	data["faxes"] = faxes
+	data["fax_id"] = fax_id
+	data["fax_name"] = fax_name
+	data["has_paper"] = paper_contain ? TRUE : FALSE
```

Comment:
this seems like an odd way to do it, why not have each fax have their own data instead?

Reply:
> this seems like an odd way to do it, why not have each fax have their own data instead?

I'm not quite sure what you mean. This generates a TGUI list of available faxes that can be accessed by the player.

Reply:
Something like this was what I had in mind:
```
/obj/machinery/fax/ui_data(mob/user)
	var/list/data = list()
	for(var/obj/machinery/fax/FAX in GLOB.machines)
		if(FAX.fax_id == fax_id) //skip yourself
			continue
		var/list/fax_data = list()

		fax_data["fax_name"] = fax_name
		fax_data["fax_id"] = fax_id
		fax_data["has_paper"] = !!paper_contain
		fax_data["syndicate_network"] = (syndicate_network || (obj_flags & EMAGGED))

		data["faxes"] += list(fax_data)

	return data
```

You can see examples of this used best in ``/datum/cellular_emporium/ui_data(mob/user)``, ``/datum/antagonist/heretic/ui_data(mob/user)``, or ``/obj/item/spellbook/ui_static_data(mob/user)``.

It requires a little more TGUI work (my example was done expecting TGUI to filter out syndicate/emagged ones, rather than DM-side) but it looks nicer (and more readable) at the end of the day.